MAINTAINERSHIP.md

```diff
@@ -0,0 +1,70 @@
+
+For contributors wishing to submit contributions to the codebase, please check our [Contributor Guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md) :)
+
+
+<p align="center">
+  <img src="maintainership.png?raw=true" alt="maintainership" width="80%" />
+</p>
+
+## Step 1: Product check
+
+Reach out to the IBC product team through susannah@interchain.io to coordinate use-case walkthrough.
+
+Answer these questions in a requirements doc:
+
+```
+    What problem does this feature solve?
+
+    What are the current solutions or workarounds? 
+
+    Are there other versions or implementations?
+
+    If there are other versions of this feature, why is this solution better?
+
+    What are the use cases?
+
+    Which users have confirmed they will use this? 
+
+    How urgent is it to implement this feature?
+
+    How soon after being developed would this be adopted? 
+
+    What is the impact of this feature being adopted?
+```
+
+Answers to these questions should also be detailed in an issue in the `ibc-go` repo to open up the discussion to a wider audience, this can be done before or after the walk through. 
+
+The acceptance criteria is based on the answers to these questions - the feature must solve a genuine problem, have users that would greatly benefit from the solution and be generic enough to benefit many users of the `ibc-go` implementation. 
+
+## Step 2: Submit spec to the IBC protocol repo
+
+A detailed review of the specification can be expected **within 2 weeks** of submission. Please notify the specification team if this does not occur, so it can be corrected as soon as possible.
+
+Any IBC code that is expected to be implemented across different chains in order to function correctly must be specified in the IBC repo to be accepted unless exempted by the specification team. 
+    
+Unilateral software (ie. code that only needs to run on a single chain to be functional) need not be submitted.
+
+If the associated module to be developed is expected to be submitted to the `ibc-go` team for maintainership, this should already be flagged at this step so that we can start thinking about/preparing our own capacity for the engineering team.
+
+## Step 3: Prepare code for handover
+
+*(this step can be initiated in parallel w/ spec submission, subject to feature complexity)*
+
+Once the spec has been given initial approval, `ibc-go` engineering will coordinate a code walkthrough in preparation for taking the module into the repo.
+
+The code that is presented should adhere to our [code contributor guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md).
+
+More details on what code walkthrough should cover will be provided by the `ibc-go` engineering team on a case by case basis. However, the code should at least be unit and E2E tested. Think about preparing for this process similarly to submitting a codebase for audit :)
```

Comment:
The code walkthrough should cover every line of the codebase in my opinion. We should understand why each line was written